calico_kubernetes/calico_kubernetes.py

```diff
@@ -328,10 +362,33 @@ def _apply_rules(self, profile_name):
             print("Error: Could not apply rules. Profile not found: %s, exiting" % profile_name)
             sys.exit(1)
 
-        rules = self._generate_rules()
-        profile_json = self._generate_profile_json(profile_name, rules)
-        profile.rules = Rules.from_json(profile_json)
-        self._datastore_client.profile_update_rules(profile)
+        inbound_rules, outbound_rules = self._generate_rules(pod)
+
+        print "Removing Default Rules"
+
+        # TODO: This method is append-only, not profile replacement, we need to remove default rules
+        # TODO: remove calicoctl calls (x5)
+        try:
+            self.calicoctl('profile', profile_name, 'rule', 'remove', 'inbound', '--at=2')
+            self.calicoctl('profile', profile_name, 'rule', 'remove', 'inbound', '--at=1')
+            self.calicoctl('profile', profile_name, 'rule', 'remove', 'outbound', '--at=1')
+        except sh.ErrorReturnCode as e:
+            print('Could not delete default rules for profile %s (assumed 2 inbound, 1 outbound)\n%s' % (profile_name, e))
+
+        for rule in inbound_rules:
+            print 'applying inbound rule \n%s' % rule
+            try:
+                self.calicoctl('profile', profile_name, 'rule', 'add', 'inbound', rule)
+            except sh.ErrorReturnCode as e:
+                print('Could not create rule %s.\n%s' % (rule, e))
```

Comment:
I also like to preface these kind of things with "ERROR" so that they are really apparent in the logs.